app/javascript/dashboard/components-next/phonenumberinput/PhoneNumberInput.vue

```diff
@@ -0,0 +1,238 @@
+<script setup>
+import { ref, computed, watch } from 'vue';
+import parsePhoneNumber from 'libphonenumber-js';
+import { useI18n } from 'vue-i18n';
+import countries from 'shared/constants/countries.js';
+import { useVuelidate } from '@vuelidate/core';
+import { required, minLength, numeric } from '@vuelidate/validators';
+import {
+  getActiveCountryCode,
+  getActiveDialCode,
+} from 'shared/components/PhoneInput/helper';
+
+import Input from 'dashboard/components-next/input/Input.vue';
+import Button from 'dashboard/components-next/button/Button.vue';
+import DropdownMenu from 'dashboard/components-next/dropdown-menu/DropdownMenu.vue';
+
+const props = defineProps({
+  modelValue: {
+    type: [String, Number],
+    default: '',
+  },
+  placeholder: {
+    type: String,
+    default: '',
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  showBorder: {
+    type: Boolean,
+    default: true,
+  },
+});
+
+const emit = defineEmits(['update:modelValue']);
+
+const { t } = useI18n();
+
+const hasInputFocused = ref(false);
+const showDropdown = ref(false);
+const searchQuery = ref('');
+const activeCountryCode = ref(getActiveCountryCode());
+const activeDialCode = ref(getActiveDialCode());
+const phoneNumber = ref(props.modelValue);
+
+const rules = {
+  phoneNumber: {
+    minLength: minLength(2),
+    numeric,
+  },
+  activeDialCode: {
+    required,
+    validDialCode: value => {
+      return countries.some(country => country.dial_code === value);
+    },
+  },
+};
+
+const v$ = useVuelidate(rules, {
+  phoneNumber,
+  activeDialCode,
+});
+
+const hasError = computed(() => v$.value.$invalid);
+
+const countryList = computed(() => {
+  return countries.map(country => ({
+    value: country.id,
+    label: country.name,
+    dialCode: country.dial_code,
+    emoji: country.emoji,
+    isSelected: String(activeCountryCode.value) === String(country.id),
+    action: 'phoneNumberInput',
+  }));
+});
+
+const filteredCountries = computed(() => {
+  const query = searchQuery.value.toLowerCase();
+  return countryList.value.filter(({ label, dialCode, value }) =>
+    [label, dialCode, value].some(field => field.toLowerCase().includes(query))
+  );
+});
+
+const activeCountry = computed(() =>
+  activeCountryCode.value
+    ? countryList.value.find(
+        country => country.value === activeCountryCode.value
+      )
+    : ''
+);
+
+const inputBorderClass = computed(() => {
+  const errorClass =
+    'border-n-ruby-8 dark:border-n-ruby-8 hover:border-n-ruby-9 dark:hover:border-n-ruby-9 disabled:border-n-ruby-8 dark:disabled:border-n-ruby-8';
+  if (!props.showBorder) {
+    return hasError.value ? errorClass : 'border-transparent';
+  }
+  if (hasError.value) {
+    return errorClass;
+  }
+  return hasInputFocused.value
+    ? 'border-n-brand dark:border-n-brand'
+    : 'border-n-weak dark:border-n-weak hover:border-n-slate-6 dark:hover:border-n-slate-6 disabled:border-n-weak dark:disabled:border-n-weak';
```

Comment:
We can use `has-[:focus] directly instead of maintain focus state in Vue context. Ref: https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-descendants